### PR TITLE
Add STSCredentials provider besides IAMCredentials provider

### PR DESCRIPTION
**Pull request by ddragosd**
This is work in progress at the moment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
 [api-gateway-hmac](https://github.com/adobe-apiplatform/api-gateway-hmac) module, and
 [lua-resty-http](https://github.com/pintsized/lua-resty-http) module.
 
-### AWS V4 Signature
-This library supports the latest AWS V4 signature which means you can use any of the latest AWS APIs without any problem.
+### AWS Credentials Provider
+Requests to AWS Services must supply valid credentials and this library provides a few credentials providers for signing AWS Requests.
+`aws_credentials` config option specifies which provider to use. 
+
+If no `aws_credentials` is provided then the library will try to guess one:
```
Comment (selfxp) [inline README.md]: If no `aws_credentials` is provided then the library will try to find one using the following order:
1. 
2.

File: README.md [hunk 2]
```diff
@@ -215,6 +268,61 @@ aws lambda get-policy --function-name hello-world-lambda-fn --region=us-east-1
...
+* Call AWS STS AssumeRole API to obtain temporary credentials.
+```lua 
+       local response, code, headers, status, body = sts:assumeRole(role_ARN,
+                role_session_name,
+                policy,
+                security_credentials_timeout,
+                external_id)       
+```
+
+More documentation on how to configure the accounts [here](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).
```
Comment (ddragosd) [inline README.md]: @stanciu I would reword like:

> For more information on how to configure the accounts see [How to Use an External ID When Granting Access to Your AWS Resources to a Third Party](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).

Pls review and change if you also agree.

File: README.md [hunk 3]
```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
...
+#### Basic Credentials
+Basic credentials work with `secret_key` and `access_key`.
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSBasicCredentials",
+    access_key = "replace-me",
+    secret_key = "replace-me"
+}
+```
+>INFO: For a better security inside the AWS environment use IAM or STS credentials.
```
Comment (selfxp) [inline README.md]: INFO: For better security

File: README.md [hunk 4]
```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
...
+    provider = "api-gateway.aws.AWSSTSCredentials",
+    role_ARN = "arn:aws:iam::111111:role/assumed_role_kinesis", -- ARN of the role to assume
+    role_session_name = "kinesis-session",                      -- a name for this session
+    shared_cache_dict = "shared_cache"                          -- shared dict for caching the credentials
+}    
+```
+Unlike IAM Credentials that exposes a single IAM Role for each EC2 instance, STS Credentials allows an EC2 instance to assume multiple roles each with its own access policy.
+
+This credentials provider uses [SecurityTokenService](src/lua/api-gateway/aws/sts/SecurityTokenService.lua) for making requests to STS and SecurityTokenService uses the IAM Credentials provider for making the call.
+ It is strongly recommended to provide the `shared_cache_dict` in order to improve performance. The temporary credentials obtained from STS are stored in it for up to 60 minutes.
```
Comment (selfxp) [inline README.md]: The temporary credentials obtained from STS are stored in `shared_cache_dict` for up to 60 minutes.

File: README.md [hunk 5]
```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
...
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSBasicCredentials",
+    access_key = "replace-me",
+    secret_key = "replace-me"
+}
+```
+>INFO: For a better security inside the AWS environment use IAM or STS credentials.
+
+### IAM Credentials
```
Comment (selfxp) [inline README.md]: Fix the indentation

File: README.md [hunk 6]
```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
...
+  This is a more secure method for signing AWS requests as credentials are short lived and the NGINX configuration doesn't need to maintain any `access_key` or `secret_key` nor worry about rotating the keys. 
+
+```lua
+aws_credentials = {
+    provider = "api-gateway.aws.AWSIAMCredentials",
+    shared_cache_dict = "my_dict"    -- the name of a shared dictionary used for caching IAM Credentials
+}
+```
+
+### STS Credentials
```
Comment (selfxp) [inline README.md]: Fix the indentation
